vmdb/app/models/container_node.rb

```diff
@@ -8,6 +8,9 @@ class ContainerNode < ActiveRecord::Base
   has_many   :container_node_conditions, :dependent => :destroy
   has_one    :computer_system, :as => :managed_entity, :dependent => :destroy
   belongs_to :lives_on, :polymorphic => true
+  has_many :containers, :through => :container_groups
+  has_many :container_services, :through => :container_groups
+  has_many :container_replicators, :through => :container_groups
```

Comment:
In all of these you have to add `:uniq => true`, e.g.:

    has_many :containers, :through => :container_groups, :uniq => true

Otherwise you would risk to fetch multiple times the same entities.